linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -4393,20 +4394,29 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
 
 private[optimizer] object OptimizerCore {
 
+  /** Hacky `LocalName` representing `this`.
+   *
+   *  `this` is normally not a valid `LocalName`, because that is a JS keyword,
+   *  so we cast here.
+   */
+  private val LocalThisName = "this".asInstanceOf[LocalName]
```

Comment:
So basically this happens because the binding infrastructure re-uses LocalName, but allows "this" as an identifier to bind to this? I need to think a bit if I can think of something better here :S

Reply:
Yes. I haven't found an easy way to address that here, but I agree we should be able to do better. I would leave that for another commit, however, as it has little to do with the current refactoring.

Reply:
Yep. SG. I take it we'll have to fix this anyways because going forward, "this" will be a valid identifier?